cmd/kubelet/app/server_linux.go

```diff
@@ -29,6 +29,7 @@ func watchForLockfileContention(path string, done chan struct{}) error {
 	}
 	if err = watcher.AddWatch(path, inotify.InOpen|inotify.InDeleteSelf); err != nil {
 		klog.ErrorS(err, "Unable to watch lockfile")
+		watcher.Close()
```

Comment:
I wonder whether this can use defer here?

Reply:
Can't use defer here, because need to return the watcher. It is closed only when an error occurs or the first event is received.

Reply:
Got it, Thanks.